packages/flutter/lib/src/material/chip.dart

```diff
@@ -10,154 +10,1447 @@ import 'package:flutter/painting.dart';
 import 'package:flutter/rendering.dart';
 import 'package:flutter/widgets.dart';
 
+import 'colors.dart';
 import 'debug.dart';
-import 'feedback.dart';
 import 'icons.dart';
+import 'ink_well.dart';
+import 'material.dart';
 import 'material_localizations.dart';
 import 'theme.dart';
 import 'tooltip.dart';
 
 // Some design constants
 const double _kChipHeight = 32.0;
 const double _kDeleteIconSize = 18.0;
-const int _kTextLabelAlpha = 0xde;
-const int _kDeleteIconAlpha = 0xde;
-const int _kContainerAlpha = 0x14;
-const double _kEdgePadding = 4.0;
+const int _kTextLabelAlpha = 0xde; // 87%
+const double _kDeleteIconOpacity = 0.87;
+const EdgeInsetsGeometry _kDefaultPadding = const EdgeInsets.all(4.0);
+const EdgeInsetsGeometry _kDefaultLabelPadding = const EdgeInsets.symmetric(horizontal: 8.0);
+const EdgeInsetsGeometry _kDefaultAvatarPadding = EdgeInsets.zero;
+const EdgeInsetsGeometry _kDefaultDeleteIconPadding = EdgeInsets.zero;
+const int _kDisabledAlpha = 0x5e; // 36%
+const double _kCheckmarkStrokeWidth = 2.0;
+const double _kPressElevation = 8.0;
+const Duration _kSelectDuration = const Duration(milliseconds: 195);
+const Duration _kCheckmarkDuration = const Duration(milliseconds: 150);
+const Duration _kCheckmarkReverseDuration = const Duration(milliseconds: 50);
+const Duration _kDrawerDuration = const Duration(milliseconds: 150);
+const Duration _kReverseDrawerDuration = const Duration(milliseconds: 100);
+const Duration _kDisableDuration = const Duration(milliseconds: 75);
+const Color _kSelectScrimColor = const Color(0x60191919);
+const Color _kDefaultSelectedColor = const Color(0x30000000); // 19% black
+const Color _kDefaultBackgroundColor = const Color(0x14000000); // 8% black
+const Color _kDefaultDisabledColor = const Color(0x08000000); // 3% black
+const Icon _kDefaultDeleteIcon = const Icon(Icons.cancel, size: _kDeleteIconSize);
+
+/// An interface defining the base attributes for a material design chip.
+///
+/// Chips are compact elements that represent an attribute, text, entity, or
+/// action.
+///
+/// The defaults mentioned in the documentation for each attribute are what
+/// the implementing classes typically use for defaults (but this class doesn't
+/// provide or enforce them).
+///
+/// See also:
+///
+///  * [Chip], a chip that displays information and can be deleted.
+///  * [InputChip], a chip that represents a complex piece of information, such
+///    as an entity (person, place, or thing) or conversational text, in a
+///    compact form.
+///  * [ChoiceChip], allows a single selection from a set of options. Choice
+///    chips contain related descriptive text or categories.
+///  * [FilterChip], uses tags or descriptive words as a way to filter content.
+///  * [ActionChip], represents an action related to primary content.
+///  * <https://material.google.com/components/chips.html>
+abstract class ChipAttributes {
```

Comment:
for classes you don't want people to extend (i.e. mixins and interfaces), include the following boilerplate code:
```dart
  // This class is intended to be used as an interface, and should not be
  // extended directly.
  factory ChipAttributes._() => null;
```